### PR TITLE
Issue #11493: Add LITERAL_FINALLY token support to WhitespaceAfter

### LINKED ISSUE #11493: WhitespaceAfter should support LITERAL_FINALLY token
https://checkstyle.org/config_whitespace.html#WhitespaceAfter

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac FinallyStatement.java
/var/tmp $

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAfter">
            <property name="tokens" value="LITERAL_FINALLY"/>
        </module>
    </module>
</module>

/var/tmp $ cat FinallyStatement.java
public class FinallyStatement {
    public static void main(String[] args) {
        try {
            System.out.println(Integer.parseInt(args[0]));
        } finally{
            System.out.println("Finished");
        }
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml FinallyStatement.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "LITERAL_FINALLY" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "LITERAL_FINALLY" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.
```

**Is your feature request related to a problem? Please describe.**
It would be nice if checkstyle could detect missing space after `finally` keyword.

**Describe the solution you'd like**
WhitespaceAfter could support new literal type - `LITERAL_FINALLY`

**Additional context**
Google Java Style Guide requires whitespace before _almost_ any `{` usage.
See https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace
>3. Before any open curly brace (`{`), with two exceptions:
> `@SomeAnnotation({a, b})` (no space is used)
> `String[][] x = {{"foo"}};` (no space is required between `{{`, by item 9 below)

#8122 seems related. This feature request is part of it.

#11489 is similar request for `try` keyword support

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Resolves #11493 
Diff Regression config: https://gist.githubusercontent.com/Rahulkhinchi03/27b23af58be50c8ac5be9e3d44db6742/raw/288d528568ed0f9959360736b0abb63ff9df14a8/config.xml
Diff Regression patch config: https://gist.githubusercontent.com/Rahulkhinchi03/f6f4d6939dd8ac952b9fce093d33ea60/raw/47b475a36fadd791c6c3fe49972771829b7feaea/patch_config.xml

### CODE REVIEW

**Comment by Rahulkhinchi03** [comment]
GitHub, generate report

**Comment by Rahulkhinchi03** [comment]
GitHub, generate site

File: src/xdocs/config_whitespace.xml [hunk 1]
```diff
@@ -2976,6 +2980,18 @@ public void myTest() {
 
     try {} catch (Exception e){} // OK
     try{} catch (Exception e){} // violation ''try' is not followed by whitespace'
+
+    try {}
+    finally { System.out.println("Hello"); } // OK
```
Comment (romani) [inline src/xdocs/config_whitespace.xml:2985]: Remove content of finally block, make it empty.
Make try-finally as single line, same as you did with try-catch

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/xdocs/config_whitespace.xml (hunk 1 above)
Comment (Rahulkhinchi03) [reply src/xdocs/config_whitespace.xml:2985]: @romani 
Line:  
`*      try {} catch (Exception e){} finally{} // violation ''finally' is not followed by whitespace'` in WhiteSpaceAfterCheck.java has more than 100 length.

Should I adjust the indentation for all or should I just remove whitespace between `finally{} //violation` and `violation''finally'`
`*      try {} catch (Exception e){} finally{}// violation''finally' is not followed by whitespace'`

Comment (Rahulkhinchi03) [reply src/xdocs/config_whitespace.xml]: I have changed `Exception --> Error` in the catch block as In WhitespaceAfterCheck.java,
line was crossing 100+ length

Comment (romani) [reply src/xdocs/config_whitespace.xml]: Good job!

**Review by romani — APPROVED** [review]
Ok to merge

**Review by pbludov — APPROVED** [review]

**Comment by pbludov** [comment]
Pitest failure for `EqualsAvoidNullCheck` is not related to this issue.

**Pull request merged**